client/matching/client.go

```diff
@@ -258,24 +258,29 @@ func (c *clientImpl) GetTaskListsByDomain(
 		futures = append(futures, future)
 	}
 
-	taskListMap := make(map[string]struct{})
+	decisionTaskListMap := make(map[string]*types.DescribeTaskListResponse)
+	activityTaskListMap := make(map[string]*types.DescribeTaskListResponse)
 	for _, future := range futures {
 		var resp *types.GetTaskListsByDomainResponse
 		if err = future.Get(ctx, &resp); err != nil {
 			return nil, err
 		}
-		for _, tl := range resp.TaskListNames {
-			if _, ok := taskListMap[tl]; !ok {
-				taskListMap[tl] = struct{}{}
+		for name, tl := range resp.GetDecisionTaskListMap() {
+			if _, ok := decisionTaskListMap[name]; !ok {
```

Comment:
if `ok`, do we need to merge the results for pollerInfo?

Reply:
Yes. It should merge the result